storm-core/src/jvm/org/apache/storm/metric/cgroup/CGroupMemoryLimit.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.metric.cgroup;
+
+import java.util.Map;
+
+/**
+ * Reports the current memory limit of the cgroup for this worker
+ */
+public class CGroupMemoryLimit extends CGroupMetricsBase<Long> {
+
+    public CGroupMemoryLimit(Map<String, Object> conf) {
+        super(conf, "memory.limit_in_bytes");
+    }
+
+    @Override
+    public Long parseFileContents(String contents) {
+        return Long.parseLong(contents.trim());
```

Comment:
I have been looking into it and it would be a lot of change just to save a few lines of code.  If you really think it is important to make the code common I will, but I would prefer to leave it.